Refine rustdoc output for do_with_in_base

dznz opened the pull request:
Documentation gardening of the following:

- Summarise the `mkHandler` function
- Describe the structs `Configuration` and `Variables`
- Annotate the public enums `Sigil` and `Escaping`
- Hide inner handlers `concatHandlerInner` and `assignmentInternalHandler` from public docs
- Hide WIP handlers `fnHandler` and `forHandler`
- Describe the type aliases `Handler` and `Handlers`

dznz committed:
docs: annotate mkHandler

Restates the contents in the Readme, but in-place.

dznz committed:
docs: hide innner, fn, for

Mark the following handler functions as hidden for documentation:

- `concatHandlerInner`
- `forHandler`
- `fnHandler`
- `assignmentInternalHandler`

The inner/internal handlers are not for use even for type 2 users, and
`for` and `fn` are WIP for now.

dznz committed:
docs: describe Escaping

Describe the enum `Escaping` options giving a couple of concrete
examples of the different escaping styles.

dznz committed:
docs: describe Configuration

Summarise the `Configuration` struct and its fields.

dznz committed:
docs: describe Variables

Summarize the `Variables` struct as holding the assignable state (as
opposed to the configurable state found in `Configuration`) for what
I'm calling a `do_with_in!` block, which isn't.. quite accurate given
the possibility of type 2 users. Words are hard.

dznz committed:
docs: describe Handler, Handlers

Try to summarise what these alias types represent (StageResult is not
pub so must talk about `Handler` in terms of the `Result` directly).

dznz committed:
docs: describe Sigil

flaviusb commented on do_with_in_base/src/lib.rs:
```diff
@@ -69,10 +70,14 @@ impl Default for Sigil {
+/// Escape sequence style set in [Configuration].
 #[derive(Debug,Copy,Clone,PartialEq,Eq)]
 pub enum Escaping {
+  /// No escape sequence is allowed.
```
It isn't that escape sequences are not allowed, it is that they don't do anything special - so eg if you have Dollar as the sigil, and `$foo` has value ⌜bar⌝, then ⌜\$foo⌝ will be processed into ⌜\bar⌝, and ⌜$$foo⌝ will be processed into ⌜$bar⌝. Though it is also worth noting that at the moment, stray ⌜\⌝ tokens explode; more work is needed in various places to make backslash escaping work.

flaviusb replied:
Github text entry field eating my comments, bleh. \\$foo will be processed into \bar

flaviusb replied:
$foo will be processed into bar

dznz committed:
docs: add summary to Escaping

The one-liners on the fields weren't quite sufficient for clarity. I was
going to add a few examples but then I couldn't decide how I wanted to
write them so I'm kicking that can down the road for now.
